### DISCUSSION TITLE
Release plan

### DISCUSSION THREAD

**Discussion by papjul** [breezy-weather/breezy-weather#6] [Announcements]
## Alpha releases (4.0.x)

* [x] Automated releases with GitHub actions (#5 in progress)

After completion, available on:
* [x] GitHub releases
* [x] Obtainium

**Goal**: by 2023-06-25

**IMPORTANT NOTE**: These alpha releases are more reliable than current Geometric Weather version for general use. However, as the name implies, this is still alpha releases. There are major changes I want to make, and I won't make migrations between updates. What can happen:
- Settings reset to default (this will happen only once when I will migrate to DataStore)
- Locations to re-add (only once after weather providers are modularized into different APK)
- Crashes just after updating. I will try to avoid this one, but if it happens, please clean app data.

For this reason, I want to avoid publishing on big stores (F-Droid, Google Play) at first, to avoid spending time debugging issues with older implementations of the code.

## Beta releases (4.1.x)

* [x] Join Weblate for translations (need cleanup first)
* [x] Export logcat from the app
* [X] Update fastlane
* Various improvements to the code to stabilize

After completion, available on:
* [X] F-Droid IzzyDroid repo (with his permission)

**Goal**: since 2023-07-08 and on IzzyOnDroid since 2023-07-11

## Stable releases (4.2.x+)

Before first stable release:
* [ ] Complete pollen for Open-Meteo
* [x] Settings are reorganized
* [X] Try to avoid making current location the default and asking for precise location permissions on first install

After completion, available on:
* [ ] F-Droid official repo

**Goal**: TBD

## Google Play Store releases (5.x+)
After getting enough feedback from F-Droid users, publish on Google Play.

* [ ] Extensions system: Open-Meteo is the default provider and other providers are available to install as different APK

Comply with Google Play policy:
* [ ] [Show privacy policy of weather providers](https://support.google.com/googleplay/android-developer/answer/9859455)
* [x] [Do not ask to turn off battery optimization](https://developer.android.com/training/monitoring-device-state/doze-standby#support_for_other_use_cases)
* [ ] [Do not ask for precise location](https://developer.android.com/privacy/best-practices#location)
* [ ] [Do not request permissions again if user already denied it](https://support.google.com/googleplay/android-developer/answer/9888170)

After completion, available on:
* [ ] Google Play Store

**Goal**: TBD

**Comment by papjul** [comment]
Won't be updated.

Please check updated milestones here: https://github.com/breezy-weather/breezy-weather/milestones

**Comment by bt4ibwem8** [comment]
I see you added your own f-droid repo. I am using Izzyondroid repo now. Does your repo have any advantages compared to Izzy repo? Should i move to your repo?

**Reply by papjul** [reply]
IzzyOnDroid contains many other apps. If you don't have any, you probably don't need such a big repo.

IzzyOnDroid is controlled by a 3rd party (Izzy) who does basic controls of what's acceptable according to some criteria in its repo. It's refreshed every day at 18:00, while mine is refreshed every time I make a release.

Our repo has pre-releases (showing as non-suggested by default) and latest stable release while IzzyOnDroid has latest two stable releases only.

Our repo is built with a CI (just like releases) so everyone can control what is being delivered.

**Reply by bt4ibwem8** [reply]
Thanks for the detailed explanation. I have some apps from Izzy repo so I'd rather stay with it because your mentioned advantages are irrelevant for me, i don't care if i get update today or tomorrow 🙂.
Btw great app, i am also translating it on Weblate.
